### PR TITLE
Monitoring: Add runbook URLs to alert and rule details pages

### PR DESCRIPTION
**Pull request by kyoto**
Jira ticket: https://issues.redhat.com/browse/MON-1649

The field is only displayed if the `runbook_url` annotation exists.

FYI @cshinn 

### Alert
alert

### Rule
rule

### CODE REVIEW

**Review by spadgett — APPROVED** [review]
/lgtm
/hold for qe-, px-, and docs-approved labels

**Comment by kyoto** [comment]
Changed "Run Book" to "Runbook"

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by lihongyan1** [comment]
Test with PR, create the following rule, can see alert detail and alert rule detail display correct Runbook
`apiVersion: monitoring.coreos.com/v1
kind: PrometheusRule
metadata:
  name: drill.rule
spec:
  groups:
  - name: Test rules
    rules:
    - alert: DrillAlert
      annotations:
        description: The is a test alert
        runbook_url: https://github.com/openshift/runbooks/blob/master/alerts/AggregatedAPIErrors.md
        message: Alerts are not configured to be sent to a notification system,
          meaning that you may not be notified in a timely fashion when important
          failures occur. Check the OpenShift documentation to learn how to configure
          notifications with Alertmanager.
      expr: vector(1)
      labels:
        severity: warning`

**Comment by kyoto** [comment]
/assign @sferich888
for px- approval

File: frontend/public/components/monitoring/alerting.tsx
Comment (sferich888) [inline frontend/public/components/monitoring/alerting.tsx:678]: What URL's are going to be in this annotation? Are these links to Red Hat KCS?

Comment (sferich888) [reply frontend/public/components/monitoring/alerting.tsx:678]: @sjstout what are docs thoughts on this. Should we / Could we have a section in docs, for these that teams could link to? thus we wouldn't need to use github links?

Comment (sferich888) [reply frontend/public/components/monitoring/alerting.tsx:678]: Ask this question this way (with a hypothetical example): If Amazon said (in its console); hey your VM is having an issue (an alert); please follow this github link (runbook) from our github org to fix the issue how would you feel (as a customer)? 

Without some 'rules' around, what links we can use here; I am leary of this PR providing customer confidence and the user experience we are looking for.

Comment (sjstout) [reply frontend/public/components/monitoring/alerting.tsx:678]: @sferich888 , I agree that pointing customers to github for official, supported product doc can be confusing. Would be happy to continue having discussions around adding this kind of content to the product doc so that customers are certain that the guidance provided is officially supported, and verified by QE.
cc: @TanyaShearonRH

Comment (spadgett) [reply frontend/public/components/monitoring/alerting.tsx:678]: cc @sichvoge

Comment (kyoto) [reply frontend/public/components/monitoring/alerting.tsx:678]: cc @brampling

Comment (jan--f) [reply frontend/public/components/monitoring/alerting.tsx:678]: The proposed plan is to export the runbook content to the openshift documentation and alter the links accordingly at release time.
I think this issue shouldn't hold up this PR in any case. I think we want to display the runbook URLs in the console and the console doesn't really have any influence on where those URLs point.
I will talk to the doc team and propose the needed alteration in CMO. I think this here can go ahead. @sferich888 any objections?

Comment (sradco) [reply frontend/public/components/monitoring/alerting.tsx:678]: Hi, suggest:
1. runbook_url should be changed based on u/s and d/s deployment. Like:
 u/s : <github link>/<runbook name>
 d/s: <d/s docs link>/<runbook name>

2. Create automatic PRs for d/s based on u/s runbooks updates .
3. In CNV we already publish the runbooks as a github pages. Currently at ```http://kubevirt.io/monitoring/``` .

Comment (jan--f) [reply frontend/public/components/monitoring/alerting.tsx:678]: yes we are hammering out a workflow for this right now. For OpenShift deployments the urls will point to the official docs, for others links to github will do.

I think this PR though should be unblocked. Iiuc this PR will display whatever link is in the actual alert. So the work needs to be done somewhere else, instead of holding up @kyoto's work for almost 6 months now.

@sferich888 Any objections against moving ahead with this PR?

**Comment by kyoto** [comment]
I've rebased this PR and also removed some refactoring in the PR that was not really related to adding the runbook links.

**Comment by kyoto** [comment]
@RickJWagner We'd like to move ahead and get this PR merged. I believe Eric's concerns have been addressed by Jan's comments, so could you please review for PX approval?

**Comment by RickJWagner** [comment]
@kyoto @jan--f 

We will be good to go (will approve with px-approved) if we have no further content on github.  

If we still have some content that's not coming from Docs, PLM considers KCS an acceptable source.  

(Just not github, we can't approve that.)

Please consider, discuss, etc.

Thanks

**Comment by RickJWagner** [comment]
Runbooks are to be put in the docs, Jan is providing scripts to swap the URLs.  PLM is happy with no content coming github.

/label px-approved

**Comment by kyoto** [comment]
Rebased
@spadgett Could you please code review this again?

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**
